# Desafio 1: Hola Mundo

In [49]:
print("Hola Mundo, esta es mi primera incursión en Python")

Hola Mundo, esta es mi primera incursión en Python


# Desafio 2: Ingreso de variable

In [50]:
name="Santiago"
edad=30
actividades=['Ciclismo','Yoga','Escalada']
mascota=False


# Desafio 3: Imprimiendo variables

In [51]:
print("Las variables creadas son: ")
print(name)
print(edad)
print(actividades[1])
print(mascota)
print("Los tipos de variables: ")
print(type(name))
print(type(edad))
print(type(actividades))
print(type(mascota))

print("Presentación")
if mascota == True:
    x=("tengo mascotas")
else:
    x=("no tengo mascotas")
print("Hola, mi nombre es {} tengo {} años, una de mis actividades favoritas es el {} y ".format(name,edad,actividades[1])+x)


Las variables creadas son: 
Santiago
30
Yoga
False
Los tipos de variables: 
<class 'str'>
<class 'int'>
<class 'list'>
<class 'bool'>
Presentación
Hola, mi nombre es Santiago tengo 30 años, una de mis actividades favoritas es el Yoga y no tengo mascotas


# Desafio 4: Codigos con errores

## Error 1
~~~~
print('Estaba la pájara pinta sentada en el verde limón)
File "/var/folders/2t/0b9zct996pqd07yvqjz4g_f80000gn/T/ipykernel_87187/3629730197.py", line 1
    print('Estaba la pájara pinta sentada en el verde limón)
                                                              ^
SyntaxError: EOL while scanning string literal
~~~~~

El error se produce debido a que no se cierra el string con una comilla simple

## Error 2

~~~~
print( 'Mi nombre es' name 'y tengo' edad, 'años' )
File "/var/folders/2t/0b9zct996pqd07yvqjz4g_f80000gn/T/ipykernel_87187/1185421031.py", line 1
    print( 'Mi nombre es' name 'y tengo' edad, 'años' )
                          ^
SyntaxError: invalid syntax
~~~~

El error se produce debido a que falta un signo + entre el el texto y la variable. Otra forma de solucionar es reemplazar la posicion de las variables con "{}" y luego agregar un .format() que contenga las variables en sus argumentos.

## Error 3 

~~~~
import padnas as pd 
import nunnpy as np
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
/var/folders/2t/0b9zct996pqd07yvqjz4g_f80000gn/T/ipykernel_87187/3541225641.py in <module>
----> 1 import padnas as pd
      2 import nunnpy as np

ModuleNotFoundError: No module named 'padnas'
~~~~

El error se produce por que no encuentra la libreria solicitada y se debe por que está mal escrita

## Error 4

~~~~
"Ornitorrinco" + 45
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/var/folders/2t/0b9zct996pqd07yvqjz4g_f80000gn/T/ipykernel_87187/1357610482.py in <module>
----> 1 "Ornitorrinco" + 45

TypeError: can only concatenate str (not "int") to str
~~~~

El error se produce dado que no se puede concatenar una variable que es de tipo str con otra variable tipo int

# Desafio 5: Utilizando Pandas y Numpy

In [66]:
import pandas as pd
import numpy as np
df=pd.read_csv('flights.csv')
df.head()


year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129
4  1949       May         121

In [53]:
df.head(5)



year     month  passengers
0  1949   January         112
1  1949  February         118
2  1949     March         132
3  1949     April         129
4  1949       May         121

In [54]:
df.tail(5)


year      month  passengers
139  1960     August         606
140  1960  September         508
141  1960    October         461
142  1960   November         390
143  1960   December         432

In [55]:
df.year.describe()

count     144.000000
mean     1954.500000
std         3.464102
min      1949.000000
25%      1951.750000
50%      1954.500000
75%      1957.250000
max      1960.000000
Name: year, dtype: float64

In [56]:
df.month.value_counts()

January      12
February     12
March        12
April        12
May          12
June         12
July         12
August       12
September    12
October      12
November     12
December     12
Name: month, dtype: int64

In [57]:
df.year.value_counts()

1949    12
1950    12
1951    12
1952    12
1953    12
1954    12
1955    12
1956    12
1957    12
1958    12
1959    12
1960    12
Name: year, dtype: int64

In [64]:
df_head_15=df.head(15)
df_head_15

year      month  passengers
0   1949    January         112
1   1949   February         118
2   1949      March         132
3   1949      April         129
4   1949        May         121
5   1949       June         135
6   1949       July         148
7   1949     August         148
8   1949  September         136
9   1949    October         119
10  1949   November         104
11  1949   December         118
12  1950    January         115
13  1950   February         126
14  1950      March         141

In [65]:
df_tail_15=df.tail(15)
df_tail_15

year      month  passengers
129  1959    October         407
130  1959   November         362
131  1959   December         405
132  1960    January         417
133  1960   February         391
134  1960      March         419
135  1960      April         461
136  1960        May         472
137  1960       June         535
138  1960       July         622
139  1960     August         606
140  1960  September         508
141  1960    October         461
142  1960   November         390
143  1960   December         432

In [59]:
passengers_mean=np.mean(df.passengers)
passengers_std=np.std(df.passengers)
passengers_median=np.median(df.passengers)

print("las estadisticas de tendencia central de df")
print("La media es {}".format(passengers_mean))
print("La mediana es {}".format(passengers_median))
print("La desviación estándar es {}".format(passengers_std))

las estadisticas de tendencia central de df
La media es 280.2986111111111
La mediana es 265.5
La desviación estándar es 119.54904153277147


## Interpretacion de las medidas de dispersión central de la base de datos flight 
Se observa que el valor promedio de los pasajeros entre 1945 y 1960 es de aproximadamente 280 por vuelo. Por otra parte la mediana que hace referencia al punto medio de los demas valores y este es de aproximadamente 265. Finalmente la desviación estandar que indica qué tan dispersos están los datos alrededor de la media es de aproximadamente 119 pasajeros. 

In [60]:
passengers_head_mean=np.mean(df_head_15.passengers)
passengers_head_std=np.std(df_head_15.passengers)
passengers_head_median=np.median(df_head_15.passengers)
print("las estadisticas de tendencia central de df_head_15")
print("La media es {}".format(passengers_head_mean))
print("La mediana es {}".format(passengers_head_median))
print("La desviación estándar es {}".format(passengers_head_std))

las estadisticas de tendencia central de df_head_15
La media es 126.8
La mediana es 126.0
La desviación estándar es 12.681745410891462


## Interpretacion de las medidas de dispersión central de la base de datos flight de las primeras 15 observaciones
Se observa que el valor promedio de los pasajeros entre *January 1949* y *March 1950* es de aproximadamente 126 por vuelo. Por otra parte la mediana que hace referencia al punto medio de los demas valores y este es de aproximadamente 126. Finalmente la desviación estandar que indica qué tan dispersos están los datos alrededor de la media es de aproximadamente 12 pasajeros. 

In [61]:
passengers_tail_mean=np.mean(df_tail_15.passengers)
passengers_tail_std=np.std(df_tail_15.passengers)
passengers_tail_median=np.median(df_tail_15.passengers)

print("las estadisticas de tendencia central de df_tail_15")
print("La media es {}".format(passengers_tail_mean))
print("La mediana es {}".format(passengers_tail_median))
print("La desviación estándar es {}".format(passengers_tail_std))

las estadisticas de tendencia central de df_tail_15
La media es 459.2
La mediana es 432.0
La desviación estándar es 75.29426715671431


## Interpretacion de las medidas de dispersión central de la base de datos flight de las ultimas 15 observaciones
Se observa que el valor promedio de los pasajeros entre *October 1959* y *December 1960* es de aproximadamente 126 por vuelo. Por otra parte la mediana que hace referencia al punto medio de los demas valores y este es de aproximadamente 126. Finalmente la desviación estandar que indica qué tan dispersos están los datos alrededor de la media es de aproximadamente 12 pasajeros. 